.github/workflows/main.yml

```diff
@@ -0,0 +1,132 @@
+name: Tbots CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  formatting-check:
+    runs-on: ubuntu-18.04
+    steps:
+      # checks-out the repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+        
+      - name: Environment Setup
+        run: |
+          "${GITHUB_WORKSPACE}"/environment_setup/setup_software.sh
+      
+      - name: Check Formatting
+        run: |
+          ./formatting_scripts/check_formatting_ci.sh
+          
+  software-tests:
+    name: Software Tests
+    runs-on: ubuntu-18.04
+    steps:
+      # checks-out the repository under $GITHUB_WORKSPACE
+      - uses: actions/checkout@v2
+
+      - name: Environment Setup
+        run: |
+          "${GITHUB_WORKSPACE}"/environment_setup/setup_software.sh
+          
+      - name: Cache Bazel External Dependencies
+        uses: actions/cache@v2
+        with:
+          path: /home/runner/.cache/bazel
+          key: bazel
+          
+      - name: Software Test
+        run: |
+          cd src
+          bazel coverage                     \
+            --compilation_mode=fastbuild     \
+            --verbose_test_summary           \
+            --instrumentation_filter=//...   \
+            --test_output=errors             \
+            -- //... -//software/simulated_tests/... \
+            -//firmware_new/...
+             
+      # uploads coverage data to codecov
+      - name: Upload Coverage
+        run: |
+         cd src
+         bash <(curl -s https://codecov.io/bash) -s bazel-testlogs/ > codecov.log
```

Comment:
Can we print the first and last 100 lines? There's something weird going on https://codecov.io/gh/UBC-Thunderbots/Software/commit/7b93cca9f6c8028b18078d4c57a80c010dfd0bc4/

Reply:
fixed